### ISSUE TITLE
to_zarr with append_dim behavior changed in 0.16.0 release 

### ISSUE THREAD

**Issue by maximemorariu** [pydata/xarray#4261]
**What happened**:
In version 0.15.1, calling `to_zarr` on a `Dataset` with a given `append_dim` would create a new zarr if it did not already exist. In version 0.16.0, this is not the case anymore: the call fails if the zarr does not exist.

**Minimal Complete Verifiable Example**:

```python
import xarray as xr

a = xr.DataArray([1, 2], {"t": [1, 2]}, ("t",))
ds = xr.Dataset({"v": a})
ds.to_zarr("CHOOSE_PATH", append_dim="t")
```

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.7 (default, May  7 2020, 21:25:33) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 4.18.0-177.el8.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8
LOCALE: en_GB.UTF-8
libhdf5: None
libnetcdf: None

xarray: 0.16.0
pandas: 1.0.5
numpy: 1.18.1
scipy: 1.5.0
netCDF4: None
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: 2.3.2
cftime: None
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 2.14.0
distributed: 2.14.0
matplotlib: None
cartopy: None
seaborn: None
numbagg: None
pint: None
setuptools: 49.2.0.post20200714
pip: 20.1.1
conda: None
pytest: 5.4.3
IPython: None
sphinx: None

</details>

**Comment by shoyer** [comment]
Interesting, I'm not quite sure how this happened! We probably didn't have test coverage.

I agree, it is convenient to be able to create a file with appending -- otherwise the first iteration of a loop needs special logic.

**Comment by maximemorariu** [comment]
So just to know, should we hope for a fix in the next release? Many thanks

**Comment by shoyer** [comment]
We would welcome a pull request to fix this, but I don't think anyone is currently working on it

**Comment by rabernat** [comment]
Fixed by #4035.
